### PR TITLE
Contact Forms

### LINKED ISSUE #81: Contact Us Form
Need to figure out where/how this works. Especially the Captcha - if it goes to AEM, we need to find that out quick.

https://www.commonmoves.com/contact-us

This block should also either export a library, or there should be a shared library to display the form and the submission logic so that it can be attached to any link on a page. 

For example, clicking this will open the pre-filled form. Same form as the contact-us page.

Contact Us

### PR DESCRIPTION
**Pull request by rrusher**
This PR focuses on the contact forms:
The contact-us page
The career inquiry form on the careers page.

Fix #81 

Test URLs:
- Before: https://main--hsf-commonmoves--hlxsites.hlx.page/contact-us
- After: https://81-form-contact--hsf-commonmoves--hlxsites.hlx.page/contact-us
- Before: https://main--hsf-commonmoves--hlxsites.hlx.page/careers#join-our-team-1
- After: https://81-form-contact--hsf-commonmoves--hlxsites.hlx.page/careers#join-our-team-1

Side modal: https://81-form-contact--hsf-commonmoves--hlxsites.hlx.page/drafts/rrusher/

### CODE REVIEW

File: scripts/delayed.js [hunk 1]
```diff
@@ -17,4 +17,27 @@ function loadAdobeLaunch() {
...
+    document.cookie = `consumerID=${idString}`;
+  }
+}
+
+if (!window.location.host.includes('localhost')) await loadAdobeLaunch();
+if (!getCookieValue('consumerID')) {
+  loadIDServlet();
+}
+
+loadScript('https://www.google.com/recaptcha/api.js?onload=onloadCallback&render=explicit', { async: true, defer: true });
```
Comment (bstopp) [inline scripts/delayed.js:43]: Do we need to load this on every page, even if we never show the form?

File: scripts/delayed.js [hunk 2]
```diff
@@ -17,4 +17,27 @@ function loadAdobeLaunch() {
...
+  const url = '/bin/bhhs/graphIdServlet';
+  const resp = await fetch(url, options);
+  if (resp.ok) {
+    const id = await resp.json();
+    const idString = JSON.stringify(id);
+    document.cookie = `consumerID=${idString}`;
+  }
+}
+
+if (!window.location.host.includes('localhost')) await loadAdobeLaunch();
```
Comment (bstopp) [inline scripts/delayed.js:38]: Still don't know why we're doing an `await` here.

File: blocks/contact-form/forms/callback.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+/* eslint-disable no-undef */
+let recaptchaToken = null;
+
+function verifyCallback(resp) {
+  recaptchaToken = resp;
+}
+
+function onloadCallback() {
+  grecaptcha.render('captcha-20285', {
+    sitekey: '6LebYaYUAAAAAC9SqASljwaF57MpKSvEkwDOzk6l',
```
Comment (bstopp) [inline blocks/contact-form/forms/callback.js:11]: This needs to be a metadata key.

File: blocks/contact-form/contact-form.js [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+          e.currentTarget.classList.add('error');
+        } else {
+          e.currentTarget.classList.remove('error');
+        }
+      });
+    });
+};
+
+export default async function decorate(block) {
+  const observer = new IntersectionObserver((entries) => {
```
Comment (bstopp) [inline blocks/contact-form/contact-form.js:373]: Do we have any pages that use the side load for testing the performance? I want to see how those are handled with this.

**Review by bstopp — CHANGES_REQUESTED** [review]

File: scripts/delayed.js (hunk 1 above)
Comment (rrusher) [reply scripts/delayed.js:43]: Moved to block.

File: scripts/delayed.js (hunk 2 above)
Comment (rrusher) [reply scripts/delayed.js:38]: removed await.

File: blocks/contact-form/forms/callback.js (hunk 3 above)
Comment (rrusher) [reply blocks/contact-form/forms/callback.js:11]: Moved to placeholders

File: blocks/contact-form/contact-form.js (hunk 4 above)
Comment (rrusher) [reply blocks/contact-form/contact-form.js:373]: Added an example in PR description.

**Review by bstopp — DISMISSED** [review]
Delta

Left is branch, right is live site

There's still some styling delta on the modal between live and branch.

And for the life of me i can't get it to open when i click on an envelope on a listing.

**Comment by rrusher** [comment]
> Delta
> 
> Left is branch, right is live site
> 
> There's still some styling delta on the modal between live and branch.
> 
> And for the life of me i can't get it to open when i click on an envelope on a listing.

I adjusted some styling around the recaptcha. This is the default form view.
Other differences are dictated by the listing. If it's featured, luxury, etc., then the form might have an agent listed instead of the office (i.e. your screenshot) as well as the address in the textarea and a disclaimer above the recaptcha.

File: blocks/fragment/fragment.css [hunk 5]
```diff
@@ -1 +1,20 @@
...
+
+.fragment-wrapper > .section:first-of-type {
+  padding-top: 0;
+}
+
+.fragment-wrapper > .section:last-of-type {
+  padding-bottom: 0;
+}
+
+.fragment-wrapper .disclaimer div p {
```
Comment (bstopp) [inline blocks/fragment/fragment.css:15]: I'm not clear why the fragment is overriding other block's CSS.

**Review by bstopp — CHANGES_REQUESTED** [review]

**Comment by rrusher** [comment]
The use of a fragment for the disclaimer copy on the Contact Us page has been removed. The copy is included in the document instead of being brought in via a Fragment. It is only being used once, this makes sense.
All other form disclaimers come from the API and will be styled via the contact-form block.

**Pull request merged**